internals/plan/plan.go

```diff
@@ -565,6 +565,11 @@ func CombineLayers(layers ...*Layer) (*Layer, error) {
 	combined.Summary = last.Summary
 	combined.Description = last.Description
 	for _, layer := range layers {
+		err := layer.Validate()
```

Comment:
Not a too bid deal, but it I see we are doing the layer checks in a place where existing layers will also be re-checked. It is a nice central place to catch them (which includes the slightly more elusive ```SetServiceArgs```

Reply:
For what it's worth, this was the case before as well, because all of these checks were previously in `CombineLayers` (unless we do move ones from `ParseLayer` as per the thread above).

@benhoyt would you prefer if the layer validation was in `ParseLayer` and `SetServiceArgs` to remove the duplication? As it is, each layer is checked every time the plan is combined even though some of the checks can't change by adding another layer.

Reply:
Just a subtle point, but I believe the previous validation was only ever done on the combined layer, once all the layers were merged. In that sense, its always needed, because the combined layer would have changed. What we are now doing is fine, but just pointing out that we are re-validating layers in the layers stack that never changed, every time we append or update an existing layer. Correct me if I am wrong, please.

Reply:
Ah, you're right, sorry. Same function, different loop :smile:.

Reply:
Yeah, I do think we should avoid validating each layer every time a new one comes in. I think we should avoid validation in `CombineLayers` altogether, and document it as not doing validation -- the caller should have validated the input layers, and will (if they need to) validate the combined output as well.